ui/packages/tidb-dashboard-for-clinic-cloud/src/apps/SlowQuery/context.ts

```diff
@@ -52,31 +55,59 @@ class DataSource implements ISlowQueryDataSource {
     timestamp?: number,
     options?: ReqConfig
   ) {
-    return client.getInstance().slowQueryDetailGet(
-      {
-        connectId,
-        digest,
-        timestamp
-      },
-      options
-    )
+    // to make this.cache as small as possible
+    const cachedItem = this.cache.pop()
+    if (cachedItem) {
+      return Promise.resolve({
+        data: cachedItem,
+        status: 200,
+        statusText: 'ok',
+        headers: {},
+        config: {}
+      })
+    } else {
+      return client.getInstance().slowQueryDetailGet(
+        {
+          connectId,
+          digest,
+          timestamp
+        },
+        options
+      )
+    }
   }
 
   slowQueryDownloadTokenPost(request: any, options?: ReqConfig) {
     return client.getInstance().slowQueryDownloadTokenPost({ request }, options)
   }
 }
 
-const ds = new DataSource()
+class EventHandler implements ISlowQueryEvent {
+  constructor(
+    public listApiReturnDetail: boolean,
+    public cache: SlowqueryModel[]
```

Comment:
In which case we need an array cache？

Reply:
You mean why the cache is array while we only need to cache one object, right? Because we need to share a same reference between DataSource and EventHandler, if we use `public cache: SlowqueryModel | undefined = undefined`, it can't be shared between DataSource and EventHandler. So I choose to use an array, Of course, map is fine as well.